packages/universal-components/src/ThemeProvider/ThemeProvider.js

```diff
@@ -0,0 +1,82 @@
+// @flow
+
+import * as React from 'react';
+
+import type { StylePropType } from '../PlatformStyleSheet/StyleTypes';
+import { createStyles } from './ThemeHelpers';
+
+export type Tokens = {|
+  fontFamilyNormal?: string,
+  fontFamilyBold?: string,
+  fontFamilyItalic?: string,
+  fontFamilyBoldItalic?: string,
+|};
+
+export type Styles = {
+  +normalFontFamily?: StylePropType,
+  +boldFontFamily?: StylePropType,
+  +italicFontFamily?: StylePropType,
+  +boldItalicFontFamily?: StylePropType,
+};
+
+export type Theme = {|
+  tokens: Tokens,
+  styles: Styles,
+|};
+
+type Props = {|
+  +children?: React.Node,
+  +tokens?: Tokens,
+|};
+
+type State = {|
+  ...Theme,
+|};
+
+const defaultTokens = {
+  fontFamilyNormal: 'Roboto',
+  fontFamilyBold: 'RobotoBold',
+  fontFamilyItalic: 'RobotoItalic',
+  fontFamilyBoldItalic: 'RobotoBoldItalic',
+};
+
+const defaultState = {
+  tokens: {
+    ...defaultTokens,
+  },
+  styles: createStyles(defaultTokens),
+};
+
+const { Provider, Consumer } = React.createContext<State>(defaultState);
+
+export default class ThemeProvider extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+
+    this.state = {
+      ...defaultState,
+    };
+  }
+
+  static getDerivedStateFromProps(props: Props) {
```

Comment:
From [React Docs](https://reactjs.org/docs/react-component.html#static-getderivedstatefromprops), it seems like `getDerivedStateFromProps` should be "avoided" if possible. Did you consider using memoization instead? It seems like it should fit this scenario.
https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html#what-about-memoization

Reply:
@RobinCsl Thanks for suggestion I wasn't worried about it. Because this wrapper should be used at the highest hierarchy level. But it's true that we cannot be sure about that.
So I updated it to use `memoize`, it also saved few lines in implementation because with this solution we don't need state anymore.

The change affected `ThemeProvider` and `ThemeHelpers` files. And I also already updated snapshots.